juju/client/connection.py

```diff
@@ -26,6 +26,8 @@
 from .facade import _JSON, _RICH_JSON, TypeEncoder
 from .facade_versions import client_facade_versions, known_unsupported_facades
 
+SPECIFIED_FACADES: TypeAlias = dict[str, dict[Literal["versions"], Sequence[int]]]
+
```

Comment:
Looks good, I'm surprised ruff doesn't complain about / fix the case - maybe because it's upper snake? https://docs.astral.sh/ruff/rules/snake-case-type-alias/

Reply:
Huh.. interesting...